testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/cdi/injection/MDBInjectionTest.java

```diff
@@ -121,19 +125,34 @@ public void testMDB() throws Exception {
         // Send a book title.
         WebTarget base = client.target(baseUri.resolve("produceMessage/"));
         String title = "Dead Man Lounging";
-        CDIInjectionBook book = new CDIInjectionBook(23, title);
-        Response response = base.request().post(Entity.entity(book, Constants.MEDIA_TYPE_TEST_XML));
-        log.trace("status: " + response.getStatus());
-        log.trace(response.readEntity(String.class));
-        Assertions.assertEquals(HttpResponseCodes.SC_OK, response.getStatus());
-        response.close();
+        try (Response response = base.request().post(Entity.entity(new CDIInjectionBook(23, title),
+                Constants.MEDIA_TYPE_TEST_XML))) {
+            log.trace("status: " + response.getStatus());
+            log.trace(response.readEntity(String.class));
+            Assertions.assertEquals(HttpResponseCodes.SC_OK, response.getStatus());
+        }
 
         // Verify that the received book title is the one that was sent.
-        base = client.target(baseUri.resolve("mdb/consumeMessage/"));
-        response = base.request().get();
-        log.trace("status: " + response.getStatus());
-        Assertions.assertEquals(HttpResponseCodes.SC_OK, response.getStatus());
-        Assertions.assertEquals(title, response.readEntity(String.class), "Wrong response");
-        response.close();
+        waitAndAssert(baseUri.resolve("mdb/consumeMessage/"), response -> {
+            log.trace("status: " + response.getStatus());
+            Assertions.assertEquals(HttpResponseCodes.SC_OK, response.getStatus());
+            Assertions.assertEquals(title, response.readEntity(String.class), "Wrong response");
+        });
+    }
+
+    private void waitAndAssert(URI uri, Consumer<Response> assertionConsumer) throws AssertionError {
+        log.info("waitAndAssert(..) validation starting.");
+        Instant endTime = Instant.now().plus(Duration.of(30, ChronoUnit.SECONDS));
+        var success = false;
+
+        while (Instant.now().isBefore(endTime) && !success) {
+            try (Response response = client.target(uri).request().get()) {
+                assertionConsumer.accept(response);
+                success = true;
+            } catch (AssertionError assertionError) {
+                log.debug("waitAndAssert(..) validation failed - retrying.");
+                Thread.onSpinWait();
+            }
+        }
```

Comment:
We need to add a line here something like:
```java
Assertions.assertTrue(success, "Did not get response within 30 seconds");
```

We probably need to catch the assertionError too and append the error message. Otherwise, we don't really know why it failed.

Reply:
Ah, right. I just realized I removed some pretty vital logic from the WildFly test I lifted this from. I've updated both methods. Hopefully that will address your concerns.